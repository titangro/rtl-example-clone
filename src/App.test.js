import { render, screen } from '@testing-library/react';
import App from './App';

describe('TEST APP', () => {
    test('renders learn react link', () => {
        render(<App />);
        const helloWorldElement = screen.getByText(/hello world!/i);
        const btn = screen.getByRole('button');
        const input = screen.getByPlaceholderText(/input value/i);
        expect(helloWorldElement).toBeInTheDocument();
        expect(btn).toBeInTheDocument();
        expect(input).toBeInTheDocument();
        // screen.debug(<App />);
        expect(input).toMatchSnapshot();
    });

    test('renders screen', () => {
        render(<App />);
        const helloWorldElement = screen.queryByText(/hello2/i);
        expect(helloWorldElement).toBeNull();
    });
});
