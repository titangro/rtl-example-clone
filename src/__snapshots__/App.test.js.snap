// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` renders learn react link 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;

exports[`TEST APP renders learn react link 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;

exports[`renders learn react link 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;
